lib/experimental/class-wp-navigation-gutenberg.php

```diff
@@ -0,0 +1,270 @@
+<?php
+/**
+ * WP_Navigation_Gutenberg class
+ *
+ * @package gutenberg
+ * @since 6.3.0
+ */
+
+/**
+ * Manages which duotone filters need to be output on the page.
+ *
+ * @access public
+ */
+class WP_Navigation_Gutenberg {
+
+	public static function get_fallback_menu() {
+
+		// 1.
+		// Get the most recently published Navigation post.
+		$navigation_post = static::get_most_recently_published_navigation();
+
+		if ( $navigation_post ) {
+			return $navigation_post;
+		}
+
+		// 2.
+		// If there are no navigation posts then try to find a classic menu
+		// and convert it into a block based navigation menu.
+		$navigation_post = static::maybe_use_classic_menu_fallback();
+
+		if ( $navigation_post ) {
+			return $navigation_post;
+		}
+
+		// 3.
+		// If there are no navigation posts then default to a list of Pages.
+		$navigation_post = static::create_default_fallback();
+
+		if ( ! is_wp_error( $navigation_post ) ) {
+			// Fetch the Post by ID
+			return get_post( $navigation_post );
```

Comment:
```suggestion
			return static::get_most_recently_published_navigation();
```

Probably more performant to use this.